src/components/ChatEntry.js

```diff
@@ -1,22 +1,42 @@
 import React from 'react';
+import { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+
+  const [color, setColor] = useState(false);
+
+  const colorClick = () => {
+    if (color === true) {
+      setColor(false)
+      props.takeLikes()
+    } else if (color === false) {
+      setColor(true)
+      props.addLikes()
+    }
+  }
```

Comment:
Once we let our single source of truth (ie, our `messages` state) handle the `liked` data, we don't need this here. Instead, this component should pass the `id` of the message whose like button was clicked to the callback, and _that_ logic should be responsible for determining what the new liked status for the message should be. This would help ensure we have a single source of truth (the list of messages), from which we can draw the entire UI.
```js
  const handleClick = () => {
    // assume the callback function below is passed in as onLikeClicked from App
    props.onLikeClicked(props.id);
  }

  // in App.js
  const handleLikeClicked = (id) => {
    // copy the list of message stored in the App state
    // copy the message indicated by the id
    // flip the liked status for the message
    // update the App state tracking the messages
  };

  // later in App.js where we pass down props to ChatLog component
  <ChatLog entries={messages} onLikeClicked={handleLikeClicked} />